test/opt/inline_test.cpp

```diff
@@ -0,0 +1,1071 @@
+       "%main = OpFunction %void None %15",
+         "%28 = OpLabel",
+         "%57 = OpVariable %_ptr_Function_float Function",
+         "%46 = OpVariable %_ptr_Function_float Function",
+         "%47 = OpVariable %_ptr_Function_float Function",
+         "%48 = OpVariable %_ptr_Function_float Function",
+      "%color = OpVariable %_ptr_Function_v4float Function",
+    "%param_1 = OpVariable %_ptr_Function_v4float Function",
+         "%29 = OpLoad %v4float %BaseColor", 
+               "OpStore %param_1 %29",
+         "%49 = OpAccessChain %_ptr_Function_float %param_1 %uint_0",
+         "%50 = OpLoad %float %49",
+         "%51 = OpAccessChain %_ptr_Function_float %param_1 %uint_1",
+         "%52 = OpLoad %float %51",
+         "%53 = OpFAdd %float %50 %52",
+               "OpStore %46 %53",
+         "%54 = OpAccessChain %_ptr_Function_float %param_1 %uint_2",
+         "%55 = OpLoad %float %54",
+               "OpStore %47 %55",
+         "%58 = OpLoad %float %46",
+         "%59 = OpLoad %float %47",
+         "%60 = OpFMul %float %58 %59",
+               "OpStore %57 %60",
+         "%56 = OpLoad %float %57",
+               "OpStore %48 %56", 
+         "%30 = OpLoad %float %48",
+         "%31 = OpCompositeConstruct %v4float %30 %30 %30 %30",
+               "OpStore %color %31",
+         "%32 = OpLoad %v4float %color",
+               "OpStore %gl_FragColor %32",
+               "OpReturn",
+               "OpFunctionEnd",
+"%foo2_f1_f1_ = OpFunction %float None %18",
+          "%f = OpFunctionParameter %_ptr_Function_float",
+         "%f2 = OpFunctionParameter %_ptr_Function_float",
+         "%33 = OpLabel",
+         "%34 = OpLoad %float %f",
+         "%35 = OpLoad %float %f2",
+         "%36 = OpFMul %float %34 %35",
+               "OpReturnValue %36",
+               "OpFunctionEnd",
+   "%foo_vf4_ = OpFunction %float None %21",
+        "%bar = OpFunctionParameter %_ptr_Function_v4float",
+         "%37 = OpLabel",
+      "%param = OpVariable %_ptr_Function_float Function",
+    "%param_0 = OpVariable %_ptr_Function_float Function",
+         "%38 = OpAccessChain %_ptr_Function_float %bar %uint_0",
+         "%39 = OpLoad %float %38",
+         "%40 = OpAccessChain %_ptr_Function_float %bar %uint_1",
+         "%41 = OpLoad %float %40",
+         "%42 = OpFAdd %float %39 %41",
+               "OpStore %param %42",
+         "%43 = OpAccessChain %_ptr_Function_float %bar %uint_2",
+         "%44 = OpLoad %float %43",
+               "OpStore %param_0 %44",
+         "%45 = OpFunctionCall %float %foo2_f1_f1_ %param %param_0",
+               "OpReturnValue %45",
+               "OpFunctionEnd",
+      // clang-format on
+  };
+  SinglePassRunAndCheck<opt::InlinePass>(JoinAllInsts(before),
+                                                 JoinAllInsts(after),
+                                                 /* skip_nop = */ false);
+}
+
+TEST_F(InlineTest, InOut) {
```

Comment:
I'd suggest to make this more clear. AFAICT, its InOutParameter, not InOutVariable.